collateral.md

```diff
@@ -26,6 +32,63 @@ module COLLATERAL
         </flips>
       </collateral>
 
+    syntax MCDContract ::= GemContract
+    syntax GemContract ::= "Gem" Address
+    syntax MCDStep ::= GemContract "." GemStep [klabel(gemStep)]
+ // ------------------------------------------------------------
+    rule contract(Gem ACCTGEM . _) => Gem ACCTGEM
+    rule [[ address(Gem ACCTGEM) => ACCTGEM ]] <gem-addr> ACCTGEM </gem-addr>
+
+    syntax GemAuthStep
+    syntax GemStep ::= GemAuthStep
+    syntax AuthStep ::= GemContract "." GemAuthStep [klabel(gemStep)]
+ // -----------------------------------------------------------------
+    rule <k> Gem _ . _ => exception ... </k> [owise]
+
+    syntax GemStep ::= "transferFrom" Address Address Int
+ // -----------------------------------------------------
+    rule <k> Gem ACCTGEM . transferFrom ACCTSRC ACCTDST VALUE => . ... </k>
+         <gem>
+           <gem-addr> ACCTGEM </gem-addr>
+           <gem-balances>...
+             ACCTSRC |-> ( BALANCE_SRC => BALANCE_SRC -Int VALUE )
+             ACCTDST |-> ( BALANCE_DST => BALANCE_DST +Int VALUE )
+           ...</gem-balances>
+         </gem>
+      requires BALANCE_SRC >=Int VALUE
+
+    syntax GemStep ::= "move" Address Address Int
+ // ---------------------------------------------
+    rule Gem ACCTGEM . (move ACCTSRC ACCTDST VALUE => transferFrom ACCTSRC ACCTDST VALUE)
+
+    syntax GemStep ::= "transfer" Address Int
+ // -----------------------------------------
+    rule <k> Gem ACCTGEM . (transfer ACCTDST VALUE => transferFrom MSGSENDER ACCTDST VALUE) ... </k>
+         <msg-sender> MSGSENDER </msg-sender>
+
+    syntax GemStep ::= "mint" Address Int
+ // -------------------------------------
+    rule <k> Gem ACCTGEM . mint ACCTDST VALUE => . ... </k>
+         <gem>
+           <gem-addr> ACCTGEM </gem-addr>
+           <gem-balances>...
+             ACCTDST |-> ( BALANCE_DST => BALANCE_DST +Int VALUE )
+           ...</gem-balances>
+         </gem>
+      requires ACCTDST =/=Int 0
+
+    syntax GemStep ::= "burn" Address Int
+ // -------------------------------------
+    rule <k> Gem ACCTGEM . burn ACCTSRC VALUE => . ... </k>
+         <gem>
+           <gem-addr> ACCTGEM </gem-addr>
+           <gem-balances>...
+             ACCTSRC |-> ( BALANCE_SRC => BALANCE_SRC -Int VALUE )
+           ...</gem-balances>
+         </gem>
+      requires ACCTSRC =/=Int 0
+       andBool BALANCE_SRC >=Int VALUE
```

Comment:
Similarly, not sure where `ACCTSRC =/=Int 0` is coming from. And we should check that `VALUE >=Int 0`.

In the solidity code for `join.sol`, for example, we check that the transfered `wad` is greater than 0 https://github.com/makerdao/dss/blob/25110e12b8323a0d1d8f45b81ebf0dbd4d740818/src/join.sol#L71